### PR TITLE
optimize clean pod log

### PR DESCRIPTION
**Pull request by KanShiori**
### What problem does this PR solve?

* Return error when deleted objects are less than input.
* Only log count for deleted objects and failed objects.
* Add log when removing backup's finalizer.

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Review by csuzhangxc — APPROVED** [review]

File: cmd/backup-manager/app/clean/clean.go [hunk 1]
```diff
@@ -60,16 +60,25 @@ func (bo *Options) cleanBRRemoteBackupData(ctx context.Context, backup *v1alpha1
...
 
 		if len(result.Deleted) != 0 {
-			klog.Infof("delete %d objects for cluster %s successfully: %s", len(result.Deleted), bo, strings.Join(result.Deleted, ","))
+			klog.Infof("For backup %s, delete %d objects successfully", bo, len(result.Deleted))
+			klog.V(4).Infof("For backup %s, all objects deleted: %s", bo, strings.Join(result.Deleted, ","))
 		}
-		if len(result.Errors) != 0 {
-			for _, oerr := range result.Errors {
-				klog.Errorf("delete object %s for cluster %s failed: %s", oerr.Key, bo, oerr.Err)
+		if len(result.Deleted) < len(objs) {
```
Comment (DanielZhangQD) [inline cmd/backup-manager/app/clean/clean.go:72]: It's possible that `len(result.Errors) != 0` when `len(result.Deleted) = len(objs)` from the code https://github.com/pingcap/tidb-operator/blob/v1.3.2/cmd/backup-manager/app/util/remote.go#L250.

Comment (KanShiori) [reply cmd/backup-manager/app/clean/clean.go]: changed, both return error

File: cmd/backup-manager/app/clean/clean.go [hunk 2]
```diff
@@ -60,17 +60,25 @@ func (bo *Options) cleanBRRemoteBackupData(ctx context.Context, backup *v1alpha1
...
 			for _, oerr := range result.Errors {
-				klog.Errorf("delete object %s for cluster %s failed: %s", oerr.Key, bo, oerr.Err)
+				klog.V(4).Infof("For backup %s, delete object %s failed: %s", bo, oerr.Key, oerr.Err)
 			}
-			return fmt.Errorf("objects remain to delete")
+			return fmt.Errorf("some objects failed to delete")
+		}
+		if len(result.Deleted) < len(objs) {
+			klog.Errorf("For backup %s, deleted objects are less than expected, deleted: %d, expected: %d", bo, len(result.Deleted), len(objs))
+			return fmt.Errorf("some objects remain to delete")
```
Comment (DanielZhangQD) [inline cmd/backup-manager/app/clean/clean.go:81]: Is it necessary to print error log and return error here?

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**